### ISSUE TITLE
Accessibility scanning with html_codesniffer as its runner fails with this error for some urls.

### ISSUE THREAD

**Issue by danghailuong1100** [pa11y/pa11y#709]
Accessibility scanning with html_codesniffer as its runner fails for [this url](https://www.transizionedigitale.it/) while it is working with axe runner.
```
Error [TypeError]: Assignment to constant variable
```
This is the pa11y config in my Node project.

```
const result = await pa11y("https://www.transizionedigitale.it/", {
      chromeLaunchConfig: {
        ignoreHTTPSErrors: false,
        args: ['--no-sandbox']
      },
      wait: 3000,
      timeout: 120000,
      standard: 'WCAG2AAA',
      includeWarnings: true,
      includeNotices: true,
      runners: ['htmlcs']
});
```

## Actual behaviour
Fails with this error:
```
Error [TypeError]: Assignment to constant variable
```
## Environment

```
System:
    OS: Linux 6.5 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
    Memory: 5.20 GB / 11.64 GB
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 21.7.1 - ~/.nvm/versions/node/v21.7.1/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v21.7.1/bin/yarn
    npm: 10.5.0 - ~/.nvm/versions/node/v21.7.1/bin/npm
  pa11y: 8.0.0
```

**Comment by malcomio** [comment]
This may not relate to html_codesniffer. Also, I think that a better description for this issue would be:

> TypeError when testing some URLs: Assignment to constant variable.

I've observed this for some URLs, just running pa11y directly from the command line.

I'm able to run pa11y for some URLs, but it fails for your example https://www.transizionedigitale.it/, and for https://www.neso.energy/ :

```
pa11y https://www.neso.energy/     

Welcome to Pa11y

 > Running Pa11y on URL https://www.neso.energy/

Error: TypeError: Assignment to constant variable.
    at h (evaluate at injectRunners (/Users/mayoung/.nvm/versions/node/v20.18.1/lib/node_modules/pa11y/lib/pa11y.js:300:20), <anonymous>:32:289113)
    at #evaluate (/Users/mayoung/.nvm/versions/node/v20.18.1/lib/node_modules/pa11y/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/ExecutionContext.js:386:56)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async ExecutionContext.evaluate (/Users/mayoung/.nvm/versions/node/v20.18.1/lib/node_modules/pa11y/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/ExecutionContext.js:273:16)
    at async IsolatedWorld.evaluate (/Users/mayoung/.nvm/versions/node/v20.18.1/lib/node_modules/pa11y/node_modules/puppeteer-core/lib/cjs/puppeteer/cdp/IsolatedWorld.js:99:16)
    at async CdpFrame.evaluate (/Users/mayoung/.nvm/versions/node/v20.18.1/lib/node_modules/pa11y/node_modules/puppeteer-core/lib/cjs/puppeteer/api/Frame.js:338:20)
    at async CdpPage.evaluate (/Users/mayoung/.nvm/versions/node/v20.18.1/lib/node_modules/pa11y/node_modules/puppeteer-core/lib/cjs/puppeteer/api/Page.js:826:20)
    at async runPa11yTest (/Users/mayoung/.nvm/versions/node/v20.18.1/lib/node_modules/pa11y/lib/pa11y.js:84:18)
    at async pa11y (/Users/mayoung/.nvm/versions/node/v20.18.1/lib/node_modules/pa11y/lib/pa11y.js:35:18)
    at async runProgram (/Users/mayoung/.nvm/versions/node/v20.18.1/lib/node_modules/pa11y/bin/pa11y.js:135:19)
```

Examples where pa11y works as expected:

```
pa11y https://www.example.com

Welcome to Pa11y

 > Running Pa11y on URL https://www.example.com

Results for URL: https://www.example.com/

 • Error: The html element should have a lang or xml:lang attribute which describes the language of the document.
   ├── WCAG2AA.Principle3.Guideline3_1.3_1_1.H57.2
   ├── html
   └── <html><head> <title>Example Domai...</html>

1 Errors
```


```
pa11y https://www.bbc.co.uk                                                                             

Welcome to Pa11y

 > Running Pa11y on URL https://www.bbc.co.uk

No issues found!
```

**Comment by aarongoldenthal** [comment]
It's really an HTML Codesniffer issue, so best to open an issue [there](https://github.com/squizlabs/HTML_CodeSniffer). If you run their bookmarklet on the page (as outlined [here](https://squizlabs.github.io/HTML_CodeSniffer/)), you get the same error (https://www.neso.energy/ in this case):

```sh
HTMLCS.js:24 Uncaught TypeError: Assignment to constant variable.
    at Object.testNullAltText (HTMLCS.js:24:223570)
    at Object.addNullAltTextResults (HTMLCS.js:24:222007)
    at Object.process (HTMLCS.js:24:221802)
    at m (HTMLCS.js:24:288483)
    at c (HTMLCS.js:24:288090)
    at Object.run (HTMLCS.js:24:287448)
    at HTMLCS.js:24:286284
    at HTMLCS.js:24:286545
    at h (HTMLCS.js:24:289153)
    at HTMLCS.js:24:289131
```

Pa11y just injects HTML Codesniffer into the page, runs it, and returns the results.

**Comment by malcomio** [comment]
Thanks - have opened https://github.com/squizlabs/HTML_CodeSniffer/issues/321
